src/NHibernate/Proxy/DefaultDynamicProxyMethodCheckerExtensions.cs

```diff
@@ -23,12 +23,10 @@ public static bool IsProxiable(this MethodInfo method)
 		public static bool ShouldBeProxiable(this MethodInfo method)
 		{
 			// to use only for real methods (no getter/setter)
-			return (method.DeclaringType != typeof (MarshalByRefObject)) &&
-			       !IsFinalizeMethod(method) &&
-			       (!(method.DeclaringType == typeof (object) && "GetType".Equals(method.Name))) &&
-			       (!(method.DeclaringType == typeof (object) && "obj_address".Equals(method.Name))) && // Mono-specific method
+			return method.DeclaringType != typeof(MarshalByRefObject) &&
+			       !(method.DeclaringType == typeof(object) && !method.IsVirtual) &&
```

Comment:
Non virtual methods of `System.Object` should not be marked as required to be proxiable. This would handle `GetType`, `obj_address` (not standard `internal` Mono method, that was removed [here](https://github.com/mono/mono/commit/4072c020e175e00c8598dcc0fa87953a6c203586)), `MemberwiseClone`, etc.